seisbench/models/base.py

```diff
@@ -62,73 +68,69 @@ def _remote_path(cls):
-        following entries, even though all arguments are optional:
+        The config file can (and should) contain the following entries, even though all arguments are optional:
 
         - "docstring": A string documenting the pipeline. Usually also contains information on the author.
         - "model_args": Argument dictionary passed to the init function of the pipeline.
         - "seisbench_requirement": The minimal version of SeisBench required to use the weights file.
         - "default_args": Default args for the :py:func:`annotate`/:py:func:`classify` functions.
           These arguments will supersede any potential constructor settings.
+        - "version": The version string of the model. For **all but the latest version**, version names should
+          furthermore be denoted in the file names, i.e., the files should end with the suffix ".v[VERSION]".
+          If no version is specified in the json, the assumed version string is "1".
+
+        .. warning::
+            Even though the version is set to "latest" by default, this will only use the latest version locally
+            available. Only if no weight is available locally, the remote repository will be queried. This behaviour
+            is implemented for privacy reasons, as it avoids contacting the remote repository for every call of the
+            function. To explicitly update to the latest version from the remote repository, set `update=True`.
 
         :param name: Model name prefix.
         :type name: str
+        :param version_str: Version of the weights to load. Either a version string or "latest". The "latest" model is
+                            the model with the highest version number.
+        :type version_str: str
         :param force: Force execution of download callback, defaults to False
         :type force: bool, optional
+        :param update: If true, downloads potential new weights file and config from the remote repository.
+                       The old files are retained with their version suffix.
+        :type update: bool
         :param wait_for_file: Whether to wait on partially downloaded files, defaults to False
         :type wait_for_file: bool, optional
         :return: Model instance
         :rtype: SeisBenchModel
         """
-        weight_path, metadata_path = cls._pretrained_path(name)
-
-        def download_weights_callback(weight_path):
-            seisbench.logger.info(f"Weight file {name} not in cache. Downloading...")
-            weight_path.parent.mkdir(exist_ok=True, parents=True)
+        cls._cleanup_local_repository()
+        if version_str == "latest":
+            versions = cls.list_versions(name, remote=update)
+            if len(versions) == 0:
+                raise ValueError(f"No version for weight '{name}' available.")
+            version_str = max(versions, key=version.parse)
```

Comment:
This is currently breaking on version lookup when trying to obtain any of the pre-trained models:


```python
ValueError                                Traceback (most recent call last)
<ipython-input-17-4adba3d07bee> in <module>
----> 1 gpd = sbm.PhaseNet.from_pretrained('stead')

~/seisbench-production/seisbench/seisbench/models/base.py in from_pretrained(cls, name, version_str, update, force, wait_for_file)
    121             versions = cls.list_versions(name, remote=update)
    122             if len(versions) == 0:
--> 123                 raise ValueError(f"No version for weight '{name}' available.")
    124             version_str = max(versions, key=version.parse)
    125 

ValueError: No version for weight 'stead' available.
```

```list_versions``` works as expected. So you get back a list of 'versions'. I think you need to change the version.parse function you pass to the ```map(...)``` call as I can't find the version variable in the method scope.

Reply:
The issue here is that by default we use `update=False`. However, this also means that we are ignoring remote versions even if **no** local version is available. I changed the behaviour to include remote versions also if no local version is available.